azurerm/internal/services/eventgrid/tests/eventgrid_system_topic_resource_test.go

```diff
@@ -0,0 +1,222 @@
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMEventGridSystemTopicDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMEventGridSystemTopic_basicWithTags(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMEventGridSystemTopicExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.foo", "bar"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "source"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "topic_type"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "metric_resource_id"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMEventGridSystemTopicDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).EventGrid.SystemTopicsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_eventgrid_system_topic" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("EventGrid System Topic still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testCheckAzureRMEventGridSystemTopicExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).EventGrid.SystemTopicsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for EventGrid System Topic: %s", name)
+		}
+
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: EventGrid System Topic %q (resource group: %s) does not exist", name, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on EventGridSystemTopicsClient: %s", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMEventGridSystemTopic_basic(data acceptance.TestData) string {
+	// TODO: confirm if this is still the case
+	// currently only supported in "West Central US" & "West US 2"
+	location := "westus2"
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```

Comment:
Could we include the service here
```suggestion
  name     = "acctestRG-eg-%d"
```